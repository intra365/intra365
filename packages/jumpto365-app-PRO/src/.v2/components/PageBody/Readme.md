



```js
<PageBody title="Page Header" color="#2a7ab9" iconUrl="https://jumpto365.com/resources/images/app/Word.png" />
```

```js
<PageBody title="Page Header no logo" color="#2a7ab9" xiconUrl="https://jumpto365.com/resources/images/app/Word.png" />
```