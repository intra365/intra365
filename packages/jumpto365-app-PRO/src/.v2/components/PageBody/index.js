import React, { Component } from 'react';
import PropTypes from 'prop-types'
import Package from "../../../../package.json"
import $ from "jquery"
/**
 * Describe overall purpose of the component
 *
 * @export
 * @class PageBody
 * @extends {Component}
 */
export default class PageBody extends Component {

    static propTypes  = {
        noscroll: PropTypes.string,
    }
    state = {}

    constructor(props) {
        super(props);
    }

    _resize = () => {
        var win = $(window);
        var height =document.documentElement.clientHeight-140; // win.height()-114;
        if (height!== this.state.height){
            this.setState({height})
        }
    
    }
 
    _onscroll = (e) => {
        //console.log("scroll",e)
    }
    componentDidMount = () => {
        this._resize()
        window.addEventListener("scroll", this._onscroll);
        window.addEventListener("resize", this._resize);
    }


    componentWillUnmount = () => {
        window.removeEventListener("scroll", this._onscroll);
        window.removeEventListener("resize", this._resize);
   
    }
    /**
     * Required method return the output of the component
     *
     * @returns
     * @memberof PageBody
     */
    render() {
        var height =  this.props.nofooter  ? this.props.height:this.state.height //this.props.height //? this.props.height : this.state.height
        var width = this.props.width ? this.props.width : "100%"
        
        return (<div>
          
            <div  style={{width,height:height,  overflowX:"hidden", overflowY:this.props.noscroll ? "hidden" : "auto"}}>

                {this.props.children}
                
</div>
{!this.props.nofooter &&
    <div style={{textAlign:"center",clear:"both",borderTop:"1px solid #cccccc",marginLeft:"-16px",marginRight:"-16px",marginTop:"8px",padding:"8px"}}>
   Copyright &copy;2017-2019 - jumpto365, Inc. - jumpto365.com&trade; - All rights reserved - <a href="https://jumpto365.com/eula/" target="_blank">Terms of Service and End User Licensing Agreement</a> version: {Package.version} {Package.subversion}
   </div>
   }</div>
        )
    }
}

