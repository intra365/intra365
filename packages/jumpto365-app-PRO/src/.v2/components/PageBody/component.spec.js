import React, { Component } from 'react';

import PageBody from '.'

test('Check that it renders', () => {

    const wrapper = shallow(<PageBody about="🍕"/>);

    expect(wrapper).toMatchSnapshot();
 
}

)