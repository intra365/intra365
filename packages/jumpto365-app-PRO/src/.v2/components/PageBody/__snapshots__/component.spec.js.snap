// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check that it renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PageBody
    about="🍕"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        <div
          style={
                    Object {
                              "borderTop": "1px solid #cccccc",
                              "clear": "both",
                              "marginLeft": "-16px",
                              "marginRight": "-16px",
                              "marginTop": "8px",
                              "padding": "8px",
                              "textAlign": "center",
                            }
          }
>
          Copyright ©2017-2018 - jumpto365, Inc. - jumpto365.com™ - All rights reserved - 
          <a
                    href="https://jumpto365.com/eula/"
                    target="_blank"
          >
                    Terms of Service and End User Licensing Agreement
          </a>
</div>,
      ],
      "style": Object {
        "height": "100%",
        "minHeight": "400px",
        "overflowX": "hidden",
        "overflowY": "auto",
        "width": "100%",
      },
    },
    "ref": null,
    "rendered": Array [
      undefined,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Copyright ©2017-2018 - jumpto365, Inc. - jumpto365.com™ - All rights reserved - ",
            <a
              href="https://jumpto365.com/eula/"
              target="_blank"
>
              Terms of Service and End User Licensing Agreement
</a>,
          ],
          "style": Object {
            "borderTop": "1px solid #cccccc",
            "clear": "both",
            "marginLeft": "-16px",
            "marginRight": "-16px",
            "marginTop": "8px",
            "padding": "8px",
            "textAlign": "center",
          },
        },
        "ref": null,
        "rendered": Array [
          "Copyright ©2017-2018 - jumpto365, Inc. - jumpto365.com™ - All rights reserved - ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Terms of Service and End User Licensing Agreement",
              "href": "https://jumpto365.com/eula/",
              "target": "_blank",
            },
            "ref": null,
            "rendered": "Terms of Service and End User Licensing Agreement",
            "type": "a",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          <div
            style={
                        Object {
                                    "borderTop": "1px solid #cccccc",
                                    "clear": "both",
                                    "marginLeft": "-16px",
                                    "marginRight": "-16px",
                                    "marginTop": "8px",
                                    "padding": "8px",
                                    "textAlign": "center",
                                  }
            }
>
            Copyright ©2017-2018 - jumpto365, Inc. - jumpto365.com™ - All rights reserved - 
            <a
                        href="https://jumpto365.com/eula/"
                        target="_blank"
            >
                        Terms of Service and End User Licensing Agreement
            </a>
</div>,
        ],
        "style": Object {
          "height": "100%",
          "minHeight": "400px",
          "overflowX": "hidden",
          "overflowY": "auto",
          "width": "100%",
        },
      },
      "ref": null,
      "rendered": Array [
        undefined,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Copyright ©2017-2018 - jumpto365, Inc. - jumpto365.com™ - All rights reserved - ",
              <a
                href="https://jumpto365.com/eula/"
                target="_blank"
>
                Terms of Service and End User Licensing Agreement
</a>,
            ],
            "style": Object {
              "borderTop": "1px solid #cccccc",
              "clear": "both",
              "marginLeft": "-16px",
              "marginRight": "-16px",
              "marginTop": "8px",
              "padding": "8px",
              "textAlign": "center",
            },
          },
          "ref": null,
          "rendered": Array [
            "Copyright ©2017-2018 - jumpto365, Inc. - jumpto365.com™ - All rights reserved - ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Terms of Service and End User Licensing Agreement",
                "href": "https://jumpto365.com/eula/",
                "target": "_blank",
              },
              "ref": null,
              "rendered": "Terms of Service and End User Licensing Agreement",
              "type": "a",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
